### PR TITLE
strict-type-predicates: warn if strictNullChecks is not enabled

### PR DESCRIPTION
**Pull request by ajafff**
#### PR checklist

- [x] Addresses an existing issue: #2783
- [ ] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:
Warn if `strict-type-predicates` is used without `strictNullChecks` and don't apply the rule. IMO that's a better behavior than just emitting wrong lint results.

This also fixes `strict-boolean-expressions` if `strict` is used instead of `strictNullChecks`.

### CODE REVIEW

File: src/language/utils.ts [hunk 1]
```diff
@@ -425,3 +425,8 @@ export function getEqualsKind(node: ts.BinaryOperatorToken): EqualsKind | undefi
             return undefined;
     }
 }
+
+export function isStrictNullChecksEnabled(options: ts.CompilerOptions): boolean {
```
Comment (ajafff) [inline src/language/utils.ts]: is this the right file for this function?

File: src/rules/strictTypePredicatesRule.ts [hunk 2]
```diff
@@ -52,6 +54,10 @@ export class Rule extends Lint.Rules.TypedRule {
     }
 
     public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
+        if (!Lint.isStrictNullChecksEnabled(program.getCompilerOptions())) {
+            showWarningOnce("strict-type-predicates does not work without strictNullChecks");
```
Comment (ajafff) [inline src/rules/strictTypePredicatesRule.ts]: Please suggest a better warning message.

File: src/language/utils.ts (hunk 1 above)
Comment (nchen63) [inline src/language/utils.ts]: I think it's okay. We'll probably have to split apart this file some time in the future

File: src/rules/strictTypePredicatesRule.ts (hunk 2 above)
Comment (nchen63) [inline src/rules/strictTypePredicatesRule.ts]: I think it's fine. Would probably be more clear if it said `tsc option --strictNullChecks`, but if you search for `strictNullChecks`, the first hit is correct

**Pull request merged**